src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -175,6 +175,15 @@ private void SetGlobalErrorIfTrue(bool arg)
 
                     methodCompiler.CompileSynthesizedMethods(privateImplClass, diagnostics);
                 }
+
+                var rootModuleType = moduleBeingBuiltOpt.RootModuleType;
+                if (rootModuleType != null)
+                {
+                    // all threads that were adding methods must be finished now, we can freeze the class:
+                    rootModuleType.Freeze();
+
+                    methodCompiler.CompileSynthesizedMethods(rootModuleType, diagnostics);
```

Comment:
>methodCompiler.CompileSynthesizedMethods(rootModuleType, diagnostics); [](start = 20, length = 70)

It feels like this is more complicated than it can be. We are talking about a single method right?

Reply:
Sure, please see my comment on this line: https://github.com/dotnet/roslyn/pull/43281#discussion_r407010880 Sounds like you don't want this kind of generality.